packages/astro/src/core/config/config.ts

```diff
@@ -165,5 +166,29 @@ export async function resolveConfig(
 		throw e;
 	}
 
+	// TODO: update examples default tsconfigs
+	if (astroConfig.experimental.typescript) {
+		const rootTsConfigUrl = new URL('./tsconfig.json', astroConfig.root);
+		if (!fsMod.existsSync(rootTsConfigUrl)) {
```

Comment:
There's a wonky thing people do sometimes where they'll intentionally not have a tsconfig in every package of their monorepo, to share a TypeScript project between them (Starlight does this, for instance)

It's super edge casey and not worth considering, imo, but still could be a problem when this is unflagged

Reply:
I'm not sure it's correct to expect the tsconfig to be beside the root here either, it can be outside of the root and upwards as typescript can be configured that way. In which case we shouldn't automatically create the `tsconfig.json` as it'll override the config set in that different place out of root.

Reply:
I think that idea was from @matthewp. Happy to remove this behavior

Reply:
Let's leave the generation of `tsconfig.json` to the user and `tsc`. They should handle that. (AKA separation of concerns)